audit/internal/scenarios/scenarios_test.go

```diff
@@ -0,0 +1,106 @@
+package scenarios
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/microsoft/go-otel-audit/audit"
+	"github.com/microsoft/go-otel-audit/audit/base"
+	"github.com/microsoft/go-otel-audit/audit/conn"
+	"github.com/microsoft/go-otel-audit/audit/msgs"
+)
+
+func TestSlowListeningMDSD(t *testing.T) {
+	t.Parallel()
+
+	const auditLogQueueSize = 1
+
+	var validRecord = msgs.Record{
+		CallerIpAddress:            msgs.MustParseAddr("192.168.0.1"),
+		CallerIdentities:           map[msgs.CallerIdentityType][]msgs.CallerIdentityEntry{msgs.UPN: {{"user1@domain.com", "Description"}}},
+		OperationCategories:        []msgs.OperationCategory{msgs.UserManagement},
+		TargetResources:            map[string][]msgs.TargetResourceEntry{"ResourceType": {{"Name", "Cluster", "DataCenter", "Region"}}},
+		CallerAccessLevels:         []string{"Level1"},
+		OperationAccessLevel:       "AccessLevel",
+		OperationName:              "Operation",
+		OperationResultDescription: "ResultDescription",
+		CallerAgent:                "Agent",
+	}
+
+	id := uuid.New().String()
+	socketName := fmt.Sprintf("/tmp/mysocket-%s.sock", id)
+	defer os.Remove(socketName)
+
+	// Create a Unix domain socket and listen for incoming connections.
+	// However, for this, we are not going to listen at all.
+	socket, err := net.Listen("unix", socketName)
+	if err != nil {
+		panic(err)
+	}
+	defer socket.Close()
+
+	// Create a function that will create a new connection to the remote audit server.
+	// We use this function to create a new connection when the connection is broken.
+	cc := func() (conn.Audit, error) {
+		return conn.NewDomainSocket(conn.DSPath(socketName))
+	}
+
+	// Creates the smart client to the remote audit server.
+	// You should only create one of these, preferrably in main().
+	c, err := audit.New(cc, audit.WithAuditOptions(base.WithSettings(base.Settings{QueueSize: auditLogQueueSize})))
+	if err != nil {
+		panic(err)
+	}
+	//defer c.Close(context.Background())
```

Comment:
Did you mean to leave this commented out?

Reply:
Not any longer.